polynote-server/src/main/scala/polynote/server/KernelPublisher.scala

```diff
@@ -203,7 +209,10 @@ class KernelPublisher private (
     _       <- versionedNotebook.update(vn => vn._1 -> vn._2.copy(path = newPath))
   } yield ()
 
-  def close(): Task[Unit] = closed.succeed(()).unit *> taskManager.shutdown()
+  def close(): TaskB[Unit] =
+    closed.succeed(()).unit *>
+      kernelRef.get.flatMap(_.fold[TaskB[Unit]](ZIO.unit)(_.shutdown())) *>
```

Comment:
Changed this to make sure we tell the kernel to close (if there is one) when shutting down the publisher.